model/common/src/icon4py/model/common/model_options.py

```diff
@@ -50,8 +75,14 @@ def setup_program(
     vertical_sizes = {} if vertical_sizes is None else vertical_sizes
     offset_provider = {} if offset_provider is None else offset_provider
 
+    if isinstance(backend, gtx.backend.Backend):
+        custom_backend = backend
+    else:
+        # customized backend params
+        custom_backend = customize_backend(**backend)
```

Comment:
We should inject the program name here, right?
```suggestion
        custom_backend = customize_backend(program = str(program.past_stage.past_node.id), **backend)
```
(or `program_name`?) and corresponding changes in `customize_backend`?